sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/ExecutorServiceParallelExecutor.java

```diff
@@ -368,25 +372,25 @@ private boolean shouldShutdown() {
     }
 
     private void mightNeedMoreWork() {
-      synchronized (scheduledExecutors) {
-        for (TransformExecutor<?> executor : scheduledExecutors.keySet()) {
-          Thread thread = executor.getThread();
-          if (thread != null) {
-            switch (thread.getState()) {
-              case BLOCKED:
-              case WAITING:
-              case TERMINATED:
-              case TIMED_WAITING:
-                break;
-              default:
-                return;
-            }
+      for (TransformExecutor<?> executor : scheduledExecutors.keySet()) {
+        Thread thread = executor.getThread();
+        if (thread != null) {
+          switch (thread.getState()) {
```

Comment:
Pull this out into

`boolean isThreadDone(Thread thread)` so that this reads:

```
Thread thread = executor.getThread();
if (!isThreadDone(thread)) {
  ...
}
```

It'll make it easier to read this method.